server/fleet/app.go

```diff
@@ -371,3 +375,120 @@ type LicenseInfo struct {
 	// Note is any additional terms of license
 	Note string `json:"note,omitempty"`
 }
+
+type Logging struct {
+	Debug           bool              `json:"debug"`
+	Json            bool              `json:"json"`
+	ResultLogPlugin string            `json:"osquery_result_log_plugin"`
+	StatusLogPlugin string            `json:"osquery_status_log_plugin"`
+	FileSystem      *FileSystemConfig `json:"filesystem,omitempty"`
+	Firehose        *FirehoseConfig   `json:"firehose,omitempty"`
+	Kinesis         *KinesisConfig    `json:"kinesis,omitempty"`
+	Lambda          *LambdaConfig     `json:"lambda,omitempty"`
+	PubSub          *PubSubConfig     `json:"pubsub,omitempty"`
+}
```

Comment:
It's possible for users to configure different log plugins for results/status and I think in that scenario this representation might be a little awkward. I wonder if it could make sense to nest the configuration details with a structure more like

```
...
"logging": {
  "result": {
    "plugin": "kinesis",
    "config": { ... },
  },
  "status": {
    "plugin": "stdout",
    "config": { ... },
  }
},
...
```

Just spitballing with that...

@chiiph what do you think?

Reply:
Given that we are only sending this as output, this makes sense, but I would try to keep it as simple as possible type-wise. Maybe like you said, and having config be `json.RawMessage`?